cnf-certification-test/lifecycle/scaling/scaling.go

```diff
@@ -152,21 +116,23 @@ func scaleDeploymentHelper(clients *clientsholder.ClientsHolder, name, namespace
 	return true
 }
 
-func runPodsetScale(clients *clientsholder.ClientsHolder, name,
+//nolint:funlen
+func runPodsetScale(podSetClient interface{}, name,
 	namespace, podsettype string, replicas int32, timeout time.Duration) error {
 	var errors error
-	switch podsettype {
-	case deployment:
-		dpClient := clients.AppsClients.Deployments(namespace)
+	switch v := podSetClient.(type) {
+	case appv1client.DeploymentInterface:
+		logrus.Infof("type is %v", v)
+		dpClient, _ := podSetClient.(appv1client.DeploymentInterface)
 		dp, err := dpClient.Get(context.TODO(), name, v1machinery.GetOptions{})
 		if err != nil {
 			logrus.Error("failed to get latest version of deployment ", namespace, ":", name)
 			return err
 		}
 		dp.Spec.Replicas = &replicas
 		_, errors = dpClient.Update(context.TODO(), dp, v1machinery.UpdateOptions{})
-	case statefulset:
-		stClient := clients.AppsClients.StatefulSets(namespace)
+	case appv1client.StatefulSetInterface:
+		stClient := podSetClient.(appv1client.StatefulSetInterface)
```

Comment:
I think you can here do 
podSetClient.Get(context.TODO(), name, v1machinery.GetOptions{}) directly. so you don't need that switch on the type. Can you check if it's possible to do so ?